Lib/test/test_poplib.py

```diff
@@ -289,6 +289,27 @@ def test_pass_(self):
     def test_stat(self):
         self.assertEqual(self.client.stat(), (10, 100))
 
+        original_shortcmd = self.client._shortcmd
+        def mock_shortcmd_invalid_format(cmd):
+            if cmd == 'STAT':
+                return b'+OK'
+            return original_shortcmd(cmd)
+
+        self.client._shortcmd = mock_shortcmd_invalid_format
+        with self.assertRaises(poplib.error_proto):
+            self.client.stat()
+
+        def mock_shortcmd_invalid_data(cmd):
+            if cmd == 'STAT':
+                return b'+OK abc def'
+            return original_shortcmd(cmd)
+
+        self.client._shortcmd = mock_shortcmd_invalid_data
+        with self.assertRaises(poplib.error_proto):
+            self.client.stat()
+
```

Comment:
You should also add a test for something like "+OK 1 2 3", to make sure that doesn't raise. See my comment above about allowing more than 3 fields in the response.

Reply:
Added the tests for case: "+OK 1 2 3". Thanks a lot